From: Christoph Hellwig <hch@lst.de>
Date: Fri, 9 Dec 2016 14:19:06 +0100
Subject: blk-wbt: don't throttle discard or write zeroes
References: bsc#1020989,FATE#322379
Git-commit: be07e14f96e3121483339a64d917fddb3b86ba98
Patch-Mainline: v4.10-rc1

Both of these are metadata only commands that are not issued by the
writeback code and not directly relevant to the writeback bandwith.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jens Axboe <axboe@fb.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 block/blk-wbt.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/block/blk-wbt.c b/block/blk-wbt.c
index d500e43..6e82769 100644
--- a/block/blk-wbt.c
+++ b/block/blk-wbt.c
@@ -575,10 +575,9 @@ static inline bool wbt_should_throttle(struct rq_wb *rwb, struct bio *bio)
 	const int op = bio_op(bio);
 
 	/*
-	 * If not a WRITE (or a discard or write zeroes), do nothing
+	 * If not a WRITE, do nothing
 	 */
-	if (!(op == REQ_OP_WRITE || op == REQ_OP_DISCARD ||
-				op == REQ_OP_WRITE_ZEROES))
+	if (op != REQ_OP_WRITE)
 		return false;
 
 	/*
-- 
1.8.5.6

